### PR TITLE
Switch to systemd networkd for all interfaces

### PR DESCRIPTION
**Pull request by mwindower**
Use systemd-networkd instead of ifupdown2 to configure interfaces even for the firewall.

This helps us with some issues:
- there is no ifupdown2 release with python-3; which is needed for Ubuntu 20.04
- currently the configuration is spread across `/etc/network/interfaces` with ifupdown2 and `/etc/systemd/network` for systemd-networkd

This also removes validating the testdata fixtures with ifup syntax-check because systemd-networkd is not able to do the same.

### CODE REVIEW

**Comment by majst01** [comment]
Should we incorporate this already in our current firewall-image overhaul PR, or should we do this in a two fold manner ?

**Comment by mwindower** [comment]
I would say this is necessary when we want to deprecate Ubuntu 19.10 and want a usable, stable 20.04 version

File: internal/netconf/interfaces.go [hunk 1]
```diff
@@ -96,59 +121,28 @@ func (v SystemdNetworkdValidator) Validate() error {
...
+	offset := 1000
+	for i, n := range kb.Networks {
 		if n.Underlay {
 			continue
 		}
 
-		e := EVPNIface{}
-		e.SVI.Comment = fmt.Sprintf("svi (networkid: %s)", n.Networkid)
-		e.SVI.VlanID = n.Vlan
+		e := Tenant{}
```
Comment (majst01) [inline internal/netconf/interfaces.go:133]: t := Tenant{}

File: internal/netconf/netobjects.go [hunk 2]
```diff
@@ -61,11 +62,12 @@ type (
 		TunnelIP string
 	}
 
-	// EVPNIface represents the information required to render EVPN interfaces configuration.
-	EVPNIface struct {
-		VRF   VRF
-		SVI   SVI
-		VXLAN VXLAN
+	// Tenant represents the information required to render EVPN interfaces configuration.
```
Comment (majst01) [inline internal/netconf/netobjects.go:65]: Why is this struct called tenant ?

File: internal/netconf/testdata/machine.yaml [hunk 3]
```diff
@@ -1,69 +1,46 @@
 ---
 hostname: machine
 networks:
-  # === Tenant Network (private=true)
```
Comment (majst01) [inline internal/netconf/testdata/machine.yaml:4]: I have found these comments quite useful

File: validate.sh [hunk 4]
```diff
@@ -17,8 +17,6 @@ validate () {
         . -t metal-networker
 }
 
-validate "ubuntu" "19.10" "7.2"
-validate "ubuntu" "19.10" "7.3"
 validate "ubuntu" "20.04" "7.2"
```
Comment (majst01) [inline validate.sh:20]: I would also remove 7.2

File: internal/netconf/interfaces.go [hunk 5]
```diff
@@ -2,92 +2,117 @@ package netconf
...
 )
 
-// NewIfacesConfigApplier constructs a new instance of this type.
-func NewIfacesConfigApplier(kind BareMetalType, kb KnowledgeBase, tmpFile string) net.Applier {
-	var data interface{}
-	var validator net.Validator
+type IfacesApplier struct {
+	kind BareMetalType
+	kb   KnowledgeBase
+	d    IfacesData
```
Comment (suckowbiz) [inline internal/netconf/interfaces.go:27]: IMHO within a function for a narrow scope it is fine to have really short names. For structs I would not apply this and choose "data" over "d".

File: internal/netconf/netobjects.go (hunk 2 above)
Comment (suckowbiz) [reply internal/netconf/netobjects.go:65]: If a strict exists to render EVPN networks I would call it like this. I would not rename it to "Tenant".

File: internal/netconf/testdata/machine.yaml [hunk 6]
```diff
@@ -1,69 +1,46 @@
...
-    # [IGNORED]
   - asn: 4200003073
-    # [IGNORED in case of private network]
     destinationprefixes: []
-    # For Machine: Used to set the loopback ips.
     ips:
       - 10.0.17.2
-    # [IGNORED in case of private network]
     nat: false
-    # [IGNORED in case of private network]
```
Comment (suckowbiz) [inline internal/netconf/testdata/machine.yaml:14]: Comments like this helps other to understand the WHY in the code. To me it is not a plus to remove those comments.

File: internal/netconf/tpl/networkd/10-lan.network.tpl [hunk 7]
```diff
@@ -1,7 +1,9 @@
-{{- /*gotype: github.com/metal-stack/metal-networker/internal/netconf.SystemdNetworkData*/ -}}
+{{- /*gotype: github.com/metal-stack/metal-networker/internal/netconf.LanNetworkData*/ -}}
```
Comment (suckowbiz) [inline internal/netconf/tpl/networkd/10-lan.network.tpl:1]: You renameed "VlanID" into "VLANID" so the consequence is to name LanNetworkData -> "LANNetworkData".

File: internal/netconf/netobjects.go (hunk 2 above)
Comment (mwindower) [reply internal/netconf/netobjects.go]: renamed it back to the original name

File: internal/netconf/interfaces.go (hunk 5 above)
Comment (mwindower) [reply internal/netconf/interfaces.go]: renamed to data

**Comment by majst01** [comment]
LGTM

**Pull request merged**